### PR TITLE
Reworked Item Groups

### PR DESCRIPTION
**Pull request by SunshineDistillery**
I reworked item groups to--hopefully--make it easier to quickly add varied item groups to new maptiles which should then automatically include any new items created in the future. 

The original file has been replaced with item_groups_primary, item_groups_gear, and item_groups_secondary.  Primary is for generic groups of stuff. Gear is for clothing and carried equipment; it's used mostly for monster drops but also places like storage lockers. Secondary contains  item groups for specific places (most of the original groups), and it mostly uses distributions which draw from groups in the former two files. I also did some various tweaking in a few other json files to keep things working.

I think mods will have to have their item groups updated, but I haven't touched them yet.  I didn't know if was Kosher to mess with other people's mods, since they're more personal than stuff in the main files.

### CODE REVIEW

**Comment by tyrael93** [comment]
Neat-

**Comment by Coolthulhu** [comment]
The new files should be in a new directory. For example, json/item_groups/gear.json, json/item_groups/food.json etc.

The PR is currently unmergeable, you need to update your master branch, merge into this PR's branch and resolve the conflict (manually, but it should be simple in this case).

**Comment by SunshineDistillery** [comment]
Yeah I can move them. I just left them in the main directory because I was worried about the load order.

**Comment by Coolthulhu** [comment]
As far as I know, item groups are supposed to be order-independent (except for modded groups, which are always loaded after core ones).

**Comment by SunshineDistillery** [comment]
I'm willing to reorganize these however people think is best, like by types of items.  Right now they're just split between the primary lists (to add new items to) and the secondary/aggregate lists (which people adding new items can ignore if they choose). I split gear off because clothing was really huge and there are thematic groups, like fireman or cop stuff, that just obviously belonged together.

**Comment by dopeghoti** [comment]
`primary` and `secondary` seem like oddly meaningless names given the scope of this change (they might as well be `apple` and `broccoli`).  I like the idea of pulling related groups into smaller files, though (see PR #13031 )  Maybe we can collaborate on this?  Right now our PRs are in direct conflict with each other.

**Comment by SunshineDistillery** [comment]
They're called "primary" and "secondary" because the primary file contains groups made of items, and the secondary file has groups that are not just lists of items, but rather are distributions of groups pulled from the primary file. So if you put stuff on a primary list, it automatically goes to the secondary lists. If that makes sense. But, yeah, those aren't very clear titles. And even splitting item_groups into three, they're still ungodly long files.  

I don't really care how the files/groups get organized. If you want to wait til this goes through, you can chop it up however you want. Or if you want to take the new item groups and stuff from this commit and incorporate them into yours. Whatever's easier or better. That second one would be fine by me because I'm really slow and terrible at using github.

**Comment by dopeghoti** [comment]
Okay, that separation makes sense.  I still have a feeling we'll probably have some overlap here, in that separating "lists of items (e. g. clothing)" and "collections of items that are used by other things (e. g. dressers which contain clothing") looks likely to naturally emerge out of the PR I'm working on.

File: Cataclysm-DDA [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit 5a939ff1a9b3d7d85cd22ae0e4cba9cd168c65a6
```
Comment (kevingranade) [inline Cataclysm-DDA]: What's this?

**Comment by kevingranade** [comment]
This sort of thing will be much more manageable if we sort out an organization scheme and then move content to separate files one piece at a time.  A 10,000 line commit is pretty much a non-starter no matter how simple what you're doing is.

**Comment by karlnp** [comment]
Hahahaha... so much for splitting up game.cpp, then.

**Comment by SunshineDistillery** [comment]
What's the best way to do it? 

Leave in the current item_groups.json, add new subdirectory with (blank) files. Then delete a group out of item_groups and move the new version of that group to the new files. Make a new commit so it's readable. Repeat.

Or maybe leave item_groups.json alone until the end and  then delete it once everything's moved?

**Comment by Coolthulhu** [comment]
Create new directory with one new file containing some of the moved groups. Possibly also move item_groups.json to the new directory.

Don't create empty files, add them once they contain something.

**Comment by dopeghoti** [comment]
Also, work in logical units, and commit small changes incrementally.  See my PR, linked above, for an example of this. In addition to being more easily able to see the evolution of the changes,  this allows you to back out one of the changes workout impacting the others.

**Pull request closed without merging**